rvip_roi_parser/src/main.cpp

```diff
@@ -0,0 +1,53 @@
+// General Includes
+#include <math.h>
+#include <cstdlib>
+#include <sstream>
+
+// Includes for ROS
+#include "ros/ros.h"
+#include "std_msgs/String.h"
+
+// Includes for ROS Image
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/image_encodings.h>
+#include <sensor_msgs/RegionOfInterest.h>
+
+// Include for darknet_ros_msgs
+#include <darknet_ros_msgs/BoundingBoxes.h>
+#include <darknet_ros_msgs/BoundingBox.h>
+
+ros::Publisher roi_pub;
+
+void roiCallback(const darknet_ros_msgs::BoundingBoxes::ConstPtr& msg)
+{
+  darknet_ros_msgs::BoundingBox input;
+  sensor_msgs::RegionOfInterest output;
+
+  for (int i = 0; i < msg->bounding_boxes.size(); i++){
+      input = msg->bounding_boxes[i];
+
+      output.x_offset = input.xmin;
+      output.y_offset = input.ymin;
+      output.width = input.xmax - input.xmin;
+      output.height = input.ymax - input.ymin;
+
+      roi_pub.publish(output);
+  }
+
+}
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "rvip_roi_parser");
+
+  ros::NodeHandle n;
+
+  // Subscription to custom YOLO bounding boxes
+  ros::Subscriber yolo_sub = n.subscribe("/darknet_ros/bounding_boxes", 1, roiCallback);
+
+  // Publication of generic sensor_msgs RegionOfInterest message
+  roi_pub = n.advertise<sensor_msgs::RegionOfInterest>("/rvip_roi_parser/roi", 1000);
```

Comment:
See earlier comment about leading slashes in topic names.

Reply:
Fixed in [`6138944`](https://github.com/ros-industrial/rvip/pull/1/commits/61389440193c7e12e18a2bac358afe3df95a29d6).

Removal of leading slash has been done to avoid instantiation of global names and improving code portability [[Reference]](https://answers.ros.org/question/212356/leading-slash-in-namespace/).